src/settings/mod.rs

```diff
@@ -403,10 +414,29 @@ impl Settings {
                     _ => None,
                 };
 
+                // 3. Virtual hosts assignment
+                let vhosts_entries = match advanced.virtual_hosts {
+                    Some(vhosts_entries) => {
+                        let mut vhosts_vec: Vec<VirtualHosts> = Vec::new();
+
+                        for vhosts_entry in vhosts_entries.iter() {
+                            if let Some(root) = vhosts_entry.root.to_owned() {
```

Comment:
Unchecked paths should throw us some runtime 404s if a root is invalid.
So we should make sure that the root path is a valid dir before appending it to the vec.
Something like https://github.com/static-web-server/static-web-server/blob/94e050b51700cbf479faef8842fec217c9b0c28d/src/server.rs#L153C33-L153C50

Reply:
Added:

```
$ cargo run -- -w tests/toml/vhost_config.toml --log-level debug
    Finished dev [unoptimized + debuginfo] target(s) in 0.14s
     Running `target/debug/static-web-server -w tests/toml/vhost_config.toml --log-level debug`
2023-08-07T22:50:18.947792Z DEBUG static_web_server::settings: toml configuration file read successfully
2023-08-07T22:50:18.948188Z DEBUG static_web_server::settings: added virtual host: example.com -> docker
2023-08-07T22:50:18.948330Z DEBUG static_web_server::server: initializing tokio runtime with multi-thread scheduler self.worker_threads=12
...
```

```
$ cargo run -- -w tests/toml/vhost_config.toml --log-level debug
    Finished dev [unoptimized + debuginfo] target(s) in 0.15s
     Running `target/debug/static-web-server -w tests/toml/vhost_config.toml --log-level debug`
2023-08-07T22:54:24.964487Z DEBUG static_web_server::settings: toml configuration file read successfully
Error: root directory for virtual host was not found or inaccessible

Caused by:
    path docker/abc was not found or inaccessible
```